### PR TITLE
feat: Windows 10 support

### PR DESCRIPTION
**Pull request by zbettenbuk**

### CODE REVIEW

File: src/BeaconDetector.mac.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import noble from '@abandonware/noble';
+
+import AbstractBeaconDetector from './AbstractBeaconDetector';
+import logger from './logger';
+import { Config, Peripherial } from './model';
+
+// Mac specific beacon device ID.
+const BEACON_DEVICE_ID = '4c000215';
```
Comment (saghul) [inline src/BeaconDetector.mac.ts]: this is a platform specific magic number, right? how does one find out the right one?

File: src/index.ts [hunk 2]
```diff
@@ -1 +1,7 @@
-export { default as BeaconDetector } from './BeaconDetector';
+import os from 'os';
+import BeaconDetectorMac from './BeaconDetector.mac';
```
Comment (saghul) [inline src/index.ts]: won't this fail at import time when the wrong platform is imported? I think TS supports dynamic imports, maybe we can solve it that way?

File: package.json [hunk 3]
```diff
@@ -36,7 +38,7 @@
     "url": "git@github.com:jitsi/spot-electron-sdk.git"
   },
   "scripts": {
-    "build": "rm -rf lib/* && tsc",
+    "build": "tsc",
```
Comment (saghul) [inline package.json]: what prompted this change?

File: README.md [hunk 4]
```diff
@@ -2,6 +2,17 @@
 
 This SDK provides spot beacon functionality for desktop (electron) apps.
 
+# Build requirements
+
+## Mac
```
Comment (saghul) [inline README.md]: macOS 😎

File: src/BeaconDetector.mac.ts (hunk 1 above)
Comment (zbettenbuk) [reply src/BeaconDetector.mac.ts]: By reverse engineering the binary data of the beacons we use

File: src/index.ts (hunk 2 above)
Comment (zbettenbuk) [reply src/index.ts]: You know, for me it's just 12 or a dozen :)

File: package.json (hunk 3 above)
Comment (zbettenbuk) [reply package.json]: Building the package on windows

Comment (saghul) [reply package.json]: Maybe we should use rimraf? https://github.com/isaacs/rimraf

File: src/index.ts [hunk 5]
```diff
@@ -1 +1,20 @@
-export { default as BeaconDetector } from './BeaconDetector';
+import os from 'os';
+
+import logger from './logger';
+
+export const beaconDetectorResolver = new Promise(async $export => {
+    let module;
+
+    switch (os.type()) {
```
Comment (saghul) [inline src/index.ts]: I've always seen checks like this performed with https://nodejs.org/api/process.html#process_process_platform not sure if there is any important difference.

**Pull request merged**